### DISCUSSION TITLE
Quay configuration file

### DISCUSSION THREAD

**Discussion by v-byte-cpu** [quay/quay#794] [General]
Hi there! Is the structure of the configuration file described somewhere to create it manually ? And is it possible to create a configuration file for quay from cli in an automated mode ?

**Comment by dmesser** [comment]
Hi, you can find the full configuration schema that users have access to here: https://docs.projectquay.io/manage_quay.html#quay-schema

You can use the Quay config editor as a graphical way to pre-create the configuration file or just assemble it in your favorite code editor and have the config editor validate it. The config editor also has an API endpoint to automate the validate, @jonathankingfc or @alecmerdler might be able to point you to the precise location.

When using the Operator you can just reference your custom configuration file in the `QuayRegistry` custom resource, see here: https://docs.projectquay.io/deploy_quay_on_openshift_op_tng.html#_customizing_the_quay_deployment

**Reply by v-byte-cpu** [reply]
Thanks! And another small question, as far as I understood the graphical editor not only generates the configuration file, but also prepares the database to run the application (triggers all migrations, etc.), am I right ?

**Reply by dmesser** [reply]
~~Yes, that's correct.~~

Actually, as of 3.4, the entire schema initialization and bootstrap logic went into the Quay container itself. So the config app is really just a way to verify your settings or generate a skeleton.

**Comment by micheelengronne** [comment]
Is it also possible to override the configuration from ENV ?

I would like to keep the ENV workflow for settings that is standard in the containers world.
